File: scripts/multi-threading/multi-threading.pl
#!/usr/bin/env perl

use v5.38;
use threads;

sub worker($t_id) {
    say "Thread $t_id started.";
    sleep(2);
    say "Thread $t_id finished.";
    return $t_id;
}

say "Process ID: $$";
say "Starting 5 threads.";

my @threads;
for my $i (1..5) {
    my $thread = threads->create(\&worker, $i);
    push @threads, $thread;
}

$_->join foreach @threads;

File: scripts/multi-threading/multi-threading.py
#!/usr/bin/env python3

import os
import time
import threading

def worker():
    name = threading.current_thread().name
    print("Thread " + name + " started.")
    time.sleep(2)
    print("Thread " + name + " finished.")

print('Process ID: ', os.getpid())
print('Starting 5 threads.')
for i in range(5):
    threading.Thread(target=worker).start()
